sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -443,33 +443,38 @@ case class InsertIntoDir(
 }
 
 /**
- * A container for holding the view description(CatalogTable), and the output of the view. The
- * child should be a logical plan parsed from the `CatalogTable.viewText`, should throw an error
- * if the `viewText` is not defined.
+ * A container for holding the view description(CatalogTable) and info whether the view is temporary
+ * or not. If the view description is available, the child should be a logical plan parsed from the
+ * `CatalogTable.viewText`. Otherwise, the view is a temporary one created from a dataframe; in this
+ * case, the child must be already resolved.
```

Comment:
hmm does it really matter? `View.child` is visible to analyzer rules and will be analyzed eventually.

Reply:
I was actually meaning when the `View` is created (to explain this require condition: `require(desc.isDefined || (isTempView && child.resolved))`)